source/common/filter/tcp_proxy.cc

```diff
@@ -277,20 +285,19 @@ void TcpProxy::onUpstreamData(Buffer::Instance& data) {
 void TcpProxy::onUpstreamEvent(Network::ConnectionEvent event) {
   if (event == Network::ConnectionEvent::RemoteClose) {
     read_callbacks_->upstreamHost()->cluster().stats().upstream_cx_destroy_remote_.inc();
-  }
-
-  if (event == Network::ConnectionEvent::LocalClose) {
-    read_callbacks_->upstreamHost()->cluster().stats().upstream_cx_destroy_local_.inc();
-  }
-
-  if (event == Network::ConnectionEvent::RemoteClose) {
     if (connect_timeout_timer_) {
       read_callbacks_->upstreamHost()->cluster().stats().upstream_cx_connect_fail_.inc();
       read_callbacks_->upstreamHost()->stats().cx_connect_fail_.inc();
     }
 
     onConnectionFailure();
-  } else if (event == Network::ConnectionEvent::Connected) {
+  }
+
+  if (event == Network::ConnectionEvent::LocalClose) {
```

Comment:
Super nitty, any reason to not leave these else-ifs?

Reply:
Sure can switch back. I was consolidating the logic and since it's impossible for any of them to happen at the same time I just made them normal ifs. I don't care though, will switch.